src/js/components/PodDetail.js

```diff
@@ -0,0 +1,234 @@
+
+import Breadcrumbs from './Breadcrumbs';
+import InternalStorageMixin from '../mixins/InternalStorageMixin';
+import MarathonStore from '../stores/MarathonStore';
+import Service from '../structs/Service';
+import ServiceActionItem from '../constants/ServiceActionItem';
+import ServiceDestroyModal from './modals/ServiceDestroyModal';
+import ServiceDetailConfigurationTab from './ServiceDetailConfigurationTab';
+import ServiceDetailDebugTab from './ServiceDetailDebugTab';
+import PodDetailInstancesTab from './PodDetailInstancesTab';
+import ServiceFormModal from './modals/ServiceFormModal';
+import ServiceSuspendModal from './modals/ServiceSuspendModal';
+import PodInfo from './PodInfo';
+import TabsMixin from '../mixins/TabsMixin';
+import VolumeTable from './VolumeTable';
+
+const METHODS_TO_BIND = [
+  'closeDialog',
+  'onActionsItemSelection',
+  'onAcceptRestartConfirmDialog',
+  'onServiceDestroyModalClose',
+  'onServiceSuspendModalClose'
+];
+
+class PodDetail extends mixin(InternalStorageMixin, StoreMixin, TabsMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {
+      instances: 'Instances',
+      configuration: 'Configuration',
+      debug: 'Debug'
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift(),
+      disabledDialog: null,
+      errorMsg: null,
+      serviceActionDialog: null
+    };
+
+    this.store_listeners = [
+      {
+        name: 'marathon',
+        events: [
+          'serviceRestartError',
+          'serviceRestartSuccess'
+        ]
+      }
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    if (this.props.mode === 'pod') {
+
+    }
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+  }
+
+  componentWillUpdate() {
+    super.componentWillUpdate(...arguments);
+  }
+
+  onActionsItemSelection(item) {
+    this.setState({serviceActionDialog: item.id});
+  }
+
+  onAcceptRestartConfirmDialog() {
+    this.setState({disabledDialog: ServiceActionItem.RESTART}, () => {
+      MarathonStore.restartService(
+        this.props.pod.id, this.shouldForceUpdate(this.state.errorMsg)
+      );
+    });
+  }
+
+  onServiceDestroyModalClose() {
+    this.closeDialog();
+    this.context.router.transitionTo('services-page');
+  }
+
+  onServiceSuspendModalClose() {
+    this.closeDialog();
+  }
+
+  onMarathonStoreServiceRestartSuccess() {
+    this.closeDialog();
+  }
+
+  onMarathonStoreServiceRestartError({message:errorMsg}) {
+    this.setState({
+      disabledDialog: null,
+      errorMsg
+    });
+  }
+
+  closeDialog() {
+    this.setState({
+      disabledDialog: null,
+      errorMsg: null,
+      serviceActionDialog: null
+    });
+  }
+
+  getRestartConfirmDialog() {
+    const {pod} = this.props;
+    const {state} = this;
+
+    return (
+      <Confirm
+        disabled={state.disabledDialog === ServiceActionItem.RESTART}
```

Comment:
personal #nit ... put this condition into a var like `isDisabled` next line as well.

I know that I am also not doing this always.